readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -653,23 +664,23 @@ def _build_pdflatex(self, tex_files):
             cmd_ret = self.build_env.run_command_class(
                 cls=latex_class,
                 cmd=cmd,
-                cwd=self.absolute_output_dir,
+                cwd=self._get_absolute_host_output_dir(),
```

Comment:
This is confusing. It was confusing to me as well.

We need to have two different paths that point to the exact same directory. How is that? The directory is mounted into a different location inside the container:

* path in the host: `/home/docs/checkouts/readthedocs.org/user_builds/<project>/`
* path in the container: `/usr/src/app/checkouts/readthedocs.org/user_builds/b9cbc24c8841/test-builds/`

Besides, the variable `$READTHEDOCS_OUTPUT` *is not defined in the host*, so we have to expand it using the full host's path. This variable cannot be used in `cwd=` due to a limitation of the Docker API (I guess) since I received an error when trying that. So, we have to fully expand it.

That said, we need:

* use the path in the host, for all the operations that are done via Python from the app (e.g. `os.path.join`, `glob.glob`, etc)
* use the path in the container, for all the operations that are executed _inside_ the container via Docker API using shell commands

Does this clarifies the situation a little at least? 🥴

Reply:
Yes. It should be a comment in the code! I'm not scared of long comments.

It's also another poke to remind us we really should be doing _everything_ in Docker, so we should prioritize that work.